src/app/screens/Transactions/index.tsx

```diff
@@ -0,0 +1,61 @@
+import Container from "@components/Container";
+import TransactionsTable from "@components/TransactionsTable";
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { toast } from "react-toastify";
+import { useSettings } from "~/app/context/SettingsContext";
+import { convertPaymentsToTransactions } from "~/app/utils/payments";
+import msg from "~/common/lib/msg";
+import { Payment, Transaction } from "~/types";
+
+function Transactions() {
+  const { t } = useTranslation("translation", {
+    keyPrefix: "transactions",
+  });
+
+  const { settings, getFormattedFiat } = useSettings();
+  const [payments, setPayments] = useState<Transaction[]>([]);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const { payments } = await msg.request<{ payments: Payment[] }>(
+        "getPayments"
+      );
+      const transactions: Transaction[] =
+        convertPaymentsToTransactions(payments);
+
+      for (const transaction of transactions) {
```

Comment:
this looks duplicated in the publishers screen. What about if convertPaymentsToTransactions took a boolean to calculate the fiat value?

Reply:
In my opinion I think it is fine to pass getFormattedFiat to convertPaymentsToTransactions. Maybe convertPaymentsToTransactions could take an options object, with getFormattedFiat being an optional function as part of the options object, and will only be called if it exists. (Rather than a boolean as I originally suggested).

But that's just my thoughts, otherwise maybe it is fine as it is. I just noticed the duplication.

Reply:
@bitcoinuser @rolznz @im-adithya @escapedcat @reneaaron Is there anything I can do to move this PR forward? From my consideration it is done. The optimization @rolznz mentioned in the last comment here (about getFormattedFiat) is the only thing left. I can implement it, I just thought I might wait for some more opinions about it. In case you approve, I'd like to finish it that it doesn't get too stale.